File: go/etl/etl.go
package etl

import "strings"

const testVersion = 1

// Transform transforms old rules format to the new one
func Transform(old map[int][]string) map[string]int {
	res := map[string]int{}
	for score, letters := range old {
		for _, letter := range letters {
			res[strings.ToLower(letter)] = score
		}
	}
	return res
}

File: clojure/etl/src/etl.clj
(ns etl
  (:require [clojure.string :as str]))

(defn transform-pair
  [new-db cost words]
  (->>
    words
    (map str/lower-case)
    (reduce #(assoc %1 %2 cost) new-db)))

(defn transform
  "Transform given values from old rules format to the new one"
  [db]
  (reduce-kv transform-pair {} db))
